src/jsonapi-datastore.js

```diff
@@ -100,6 +101,17 @@ class JsonApiDataStore {
    * @param {object} model The model to destroy.
    */
   destroy(model) {
+    this.graph[model._type][model.id]._associations.map(function(assoc) {
```

Comment:
No reason, though you pointing it out, it does make sense to change it to a forEach.